### PR TITLE
Automatically jump you to your current room if you are on DICE

### PR DESCRIPTION
**Pull request by qaisjp**
new mapp feature: when you log in, if you are on a dice machine, your default page will be wherever you currently are (assuming mapp knows where that machine is)
(does that one by getting the machine name from your IP address, not by tracking where you are)

### CODE REVIEW

File: website/map/views.py [hunk 1]
```diff
@@ -232,6 +233,19 @@ def index():
...
+        hostname, _, _ = socket.gethostbyaddr(request.remote_addr)
+
+        suffix = ".inf.ed.ac.uk"
+        if hostname.endswith(suffix):
+            machine = hostname[:-len(suffix)]
+            if machine.isalpha() and (flask_redis.exists(machine) == 1):
+                r = flask_redis.hget(machine, "room")
+                return redirect("/site/"+r)
+
+    except socket.herror:
```
Comment (neanias) [inline website/map/views.py:246]: What can cause this error?

Comment (qaisjp) [reply website/map/views.py:246]: unknown hosts

https://docs.python.org/2/library/socket.html#socket.herror

Comment (neanias) [reply website/map/views.py:246]: `s/2/3/`

Comment (qaisjp) [reply website/map/views.py:246]: same

**Review by neanias — APPROVED** [review]
LGTM :shipit:

**Pull request merged**